File: src/CustomerBuilder.java
public class CustomerBuilder implements CustomerBuilderInterface{

    private String firstName;
    private String middleName;
    private String lastName;
    private String primaryEmail;
    private String secondaryEmail;
    private String mobile;
    private String address;



    @Override
    public CustomerBuilder firstName(String firstname) {
        this.firstName = firstname;
        return this;
    }

    @Override
    public CustomerBuilder middleName(String middleName) {
        this.middleName = middleName;
        return this;
    }

    @Override
    public CustomerBuilder lastName(String lastName) {
        this.lastName = lastName;
        return this;
    }

    @Override
    public CustomerBuilder primaryEmail(String primaryEmail) {
        this.primaryEmail = primaryEmail;
        return this;
    }

    @Override
    public CustomerBuilder secondaryEmail(String secondaryEmail) {
        this.secondaryEmail = secondaryEmail;
        return this;
    }

    @Override
    public CustomerBuilder mobile(String mobile) {
        this.mobile = mobile;
        return this;
    }

    @Override
    public CustomerBuilder address(String address) {
        this.address = address;
        return this;
    }

    public Customer build(){
        return new Customer(this);
    }

    public String getFirstName() {
        return firstName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getPrimaryEmail() {
        return primaryEmail;
    }

    public String getSecondaryEmail() {
        return secondaryEmail;
    }

    public String getMobile() {
        return mobile;
    }

    public String getAddress() {
        return address;
    }

}

File: src/CustomerBuilder.kt
class CustomerBuilder : CustomerBuilderInterface{

     private var firstName : String = ""
     private var middleName : String = ""
     private var lastName : String = ""
     private var primaryEmail : String = ""
     private var secondaryEmail : String = ""
     private var mobile : String = ""
     private var address : String = ""




    override fun firstName(firstname : String) = apply {
        this.firstName = firstname
        return this
    }


    override fun middleName(middleName : String) = apply{
        this.middleName = middleName
        return this
    }


    override fun lastName(lastName : String) = apply{
        this.lastName = lastName
        return this
    }

    override fun primaryEmail(primaryEmail : String) = apply{
        this.primaryEmail = primaryEmail
        return this
    }


    override fun secondaryEmail(secondaryEmail : String) = apply{
        this.secondaryEmail = secondaryEmail
        return this
    }


    override fun mobile(mobile : String) = apply{
        this.mobile = mobile
        return this
    }


    override fun address(address : String) = apply{
        this.address = address
        return this
    }

    fun build() : Customer{
        return Customer(this)
    }

    fun getFirstName() : String{
        return firstName
    }

    fun getMiddleName() : String{
        return middleName
    }

    fun getLastName() : String{
        return lastName
    }

    fun getPrimaryEmail() : String{
        return primaryEmail
    }

    fun getSecondaryEmail() : String{
        return secondaryEmail
    }

    fun getMobile() : String{
        return mobile
    }

    fun getAddress() : String{
        return address
    }

}
